src/Dashboard/Grid.php

```diff
@@ -279,7 +276,16 @@ public function show(bool $mini = false)
             }
         }
 
-       // prepare all available cards
+        // Force clear the cards cache everytime the page is loaded
+        // This is needed because some plugins might add cards at runtime (like
+        // advanceddashboards) and are not aware of this new cache
+        // In the future (TODO 10.1 ?) plugins should clear this cache entry
+        // themselves as needed
+        $cache_manager = new CacheManager();
+        $GLPI_CACHE = $cache_manager->getCoreCacheInstance();
+        $GLPI_CACHE->delete('getAllDasboardCards');
```

Comment:
1. `$GLPI_CACHE` should be accessed from its global declaration.
2. If we clean this cache entry everytime a plugin is activated/deactivated, there is no reason to clear this entry on each page (except on debug mode, as it can correspond to a "dev" environment).

```suggestion
        // In debug mode, force clear the cards cache everytime the page is loaded.
        // This is needed because some plugins might add cards at runtime (like
        // advanceddashboards) and are not aware of this new cache
        //
        // FIXME: When we will be able to distinguish "dev" and "prod" environments,
        // do this only in "dev" context.
        global $GLPI_CACHE;
        if ($_SESSION['glpi_use_mode'] === Session::DEBUG_MODE) {
            $GLPI_CACHE->delete('getAllDasboardCards');
        }
```

Reply:
> If we clean this cache entry everytime a plugin is activated/deactivated, there is no reason to clear this entry on each page (except on debug mode, as it can correspond to a "dev" environment).

The plugin `advanceddashboard` can add new cards at glance (after its installation)

Reply:
Yes that's what I tried to imply in the big comment block.
We can reword the comment if that wasn't clear for you @cedric-anne ?

Reply:
Solution could be to store into GLPI core cache the core cards only and delegate to each plugin the responsibility of implementing a caching logic.
```php
$core_cards = $GLPI_CACHE->get('getAllDasboardCards');
if ($core_cards === null || $force) {
   // build cards
}

$GLPI_CACHE->set('getAllDasboardCards');

$more_cards = Plugin::doHookFunction(Hooks::DASHBOARD_CARDS);
$cards = is_array($more_cards) ? array_merge($core_cards, $more_cards) : $core_cards;
```

Reply:
we discussed that with @AdrienClairembault.
And for a bugfixes, i'm not in favor of this solution.
I took the example of advanceddashboard, but maybe other plugin do the same.

And for one call, we don't lost so much time.
So the current solution is ok i think

Reply:
Done [6f6044b](https://github.com/glpi-project/glpi/pull/12973/commits/6f6044b0c9df97d86558fdff2ebbf90abae661de).